framework/test/unit/modules/nft/commands/transfer.spec.ts

```diff
@@ -0,0 +1,294 @@
+		});
+
+		it('should fail if recipientAddress is not 20 bytes', async () => {
+			const recipientAddressIncorrectLengthContext = createTransactionContextWithOverridingParams({
+				recipientAddress: utils.getRandomBytes(22),
+			});
+
+			await expect(
+				command.verify(
+					recipientAddressIncorrectLengthContext.createCommandVerifyContext(transferParamsSchema),
+				),
+			).rejects.toThrow("'.recipientAddress' address length invalid");
+		});
+
+		it('should fail if data exceeds 64 characters', async () => {
+			const dataIncorrectLengthContext = createTransactionContextWithOverridingParams({
+				data: '1'.repeat(65),
+			});
+
+			await expect(
+				command.verify(dataIncorrectLengthContext.createCommandVerifyContext(transferParamsSchema)),
+			).rejects.toThrow("'.data' must NOT have more than 64 characters");
+		});
+
+		it('should fail if nftID does not exist', async () => {
+			const nftIDNotExistingContext = createTransactionContextWithOverridingParams({
+				nftID: Buffer.alloc(LENGTH_NFT_ID, 0),
+			});
+
+			await expect(
+				command.verify(nftIDNotExistingContext.createCommandVerifyContext(transferParamsSchema)),
+			).rejects.toThrow('NFT substore entry does not exist');
+		});
+
+		it('should fail if NFT is escrowed to another chain', async () => {
+			const escrowedNFTID = utils.getRandomBytes(LENGTH_NFT_ID);
+			const chainID = utils.getRandomBytes(LENGTH_CHAIN_ID);
+
+			const nftEscrowedContext = createTransactionContextWithOverridingParams({
+				nftID: escrowedNFTID,
+			});
+
+			const nftStore = module.stores.get(NFTStore);
+			await nftStore.set(createStoreGetter(nftEscrowedContext.stateStore), escrowedNFTID, {
+				owner: chainID,
+				attributesArray: [],
+			});
+
+			await expect(
+				command.verify(nftEscrowedContext.createCommandVerifyContext(transferParamsSchema)),
+			).rejects.toThrow('NFT is escrowed to another chain');
+		});
+
+		it('should fail if owner of the NFT is not the sender', async () => {
+			const nftID = utils.getRandomBytes(LENGTH_NFT_ID);
+
+			const nftIncorrectOwnerContext = createTransactionContextWithOverridingParams({
+				nftID,
+			});
+
+			const nftStore = module.stores.get(NFTStore);
+			await nftStore.save(createStoreGetter(nftIncorrectOwnerContext.stateStore), nftID, {
+				owner: utils.getRandomBytes(LENGTH_ADDRESS),
+				attributesArray: [],
+			});
+
+			await expect(
+				command.verify(nftIncorrectOwnerContext.createCommandVerifyContext(transferParamsSchema)),
+			).rejects.toThrow('Transfer not initiated by the NFT owner');
+		});
+
+		it('should fail if NFT exists and is locked by its owner', async () => {
+			const nftID = utils.getRandomBytes(LENGTH_NFT_ID);
+			const senderPublicKey = utils.getRandomBytes(32);
+			const ownerID = address.getAddressFromPublicKey(senderPublicKey);
+
+			const lockedNFTContext = createTransactionContextWithOverridingParams(
+				{ nftID },
+				{ senderPublicKey },
+			);
+
+			const nftStore = module.stores.get(NFTStore);
```

Comment:
common code can be moved to local `beforeEach` instead of re-declaring in multiple cases. same for other places where applicable.

Reply:
[Modified under the commit](https://github.com/LiskHQ/lisk-sdk/pull/8503/commits/7251b6180ba89a0a7575cfff6a53289971f936c4)